source/postprocess/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,833 @@
+      return Output::not_found;
+    }
+
+
+
+    template <int dim>
+    void
+    CrystalPreferredOrientation<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Crystal Preferred Orientation");
+        {
+          prm.declare_entry ("Time between data output", "1e8",
+                             Patterns::Double (0),
+                             "The time interval between each generation of "
+                             "output files. A value of zero indicates that "
+                             "output should be generated every time step.\n\n"
+                             "Units: years if the "
+                             "'Use years in output instead of seconds' parameter is set; "
+                             "seconds otherwise.");
+
+          prm.declare_entry ("Random number seed", "1",
+                             Patterns::Integer (0),
+                             "The seed used to generate random numbers. This will make sure that "
+                             "results are reproducable as long as the problem is run with the "
+                             "same amount of MPI processes. It is implemented as final seed = "
+                             "random number seed + MPI Rank. ");
+
+          prm.declare_entry ("Write in background thread", "false",
+                             Patterns::Bool(),
+                             "File operations can potentially take a long time, blocking the "
+                             "progress of the rest of the model run. Setting this variable to "
+                             "`true' moves this process into background threads, while the "
+                             "rest of the model continues.");
+
+          prm.declare_entry ("Temporary output location", "",
+                             Patterns::Anything(),
+                             "On large clusters it can be advantageous to first write the "
+                             "output to a temporary file on a local file system and later "
+                             "move this file to a network file system. If this variable is "
+                             "set to a non-empty string it will be interpreted as a "
+                             "temporary storage location.");
+
+          prm.declare_entry ("Write out raw cpo data",
+                             "olivine volume fraction,olivine Euler angles,enstatite volume fraction,enstatite Euler angles",
+                             Patterns::List(Patterns::Anything()),
+                             "A list containing the what part of the particle cpo data needs "
+                             "to be written out after the particle id. This writes out the raw "
+                             "cpo data files for each MPI process. It can write out the following data: "
+                             "olivine volume fraction, olivine A matrix, olivine Euler angles, "
```

Comment:
here you list three options for both enstatite and olivine, but in line 611 there are only 2 options per mineral (the A matrix does not exist as option). Why is that?

Reply:
There are 3 options, but the A matrix was changed to rotation matrix. I have updated the names in the documents now.